### PR TITLE
feat(curriculum): date conversion lab

### PR DESCRIPTION
**Pull request by zairahira**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

I have slightly modified the original prototype and have removed `console.log` statements from the functions.

Original prototype: https://github.com/freeCodeCamp/CurriculumExpansion/pull/500/

Closes #XXXXX

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+You should have a variable named `currentDate` that holds the current date and time using the `Date` object.
+
+```js
+
+// excluded timestamps
+const expectedDate = new Date();
+const expectedDateOnly = new Date(expectedDate.getFullYear(), expectedDate.getMonth(), expectedDate.getDate());
+
+const currentDateOnly = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate());
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md:37]: why did you create these but did not use them?

File: curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+You should have a function `formatDateLong` 
+
+```js
+assert.isFunction(formatDateLong);
+```
+
+The function`formatDateLong` should take a date object as a parameter.
+
+```js
+assert.match(formatDateLong.toString(), /\s*formatDateLong\(\s*\w+\s*\)/);
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md:94]: how is this testing that the parameter is a date object?

File: curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+```js
+
+// excluded timestamps
+const expectedDate = new Date();
+const expectedDateOnly = new Date(expectedDate.getFullYear(), expectedDate.getMonth(), expectedDate.getDate());
+
+const currentDateOnly = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate());
+
+assert.equal(currentDate.toString(), expectedDate.toString());
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md:39]:
```suggestion
assert.equal(currentDateOnly.toString(), expectedDateOnly.toString());
```
I guess this was the idea.

File: curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+```js
+assert.isFunction(formatDateMMDDYYYY);
+```
+
+The function`formatDateMMDDYYYY` should take a date object as a parameter.
+
+```js
+assert.match(formatDateMMDDYYYY.toString(), /\s*formatDateMMDDYYYY\(\s*\w+\s*\)/);
+```
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md:67]: this is not testing that the parameter is a date object. To test if it has one parameter you can check `formatDateMMDDYYY.length` (I learned recently that the lenght is equal to the number of parameters)

````suggestion
The function`formatDateMMDDYYYY` should take one parameter.

```js
assert.lengthOf(formatDateMMDDYYYY, 1);
```
````

File: curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md (hunk 2 above)
Comment (ilenia-magoni) [reply curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md:94]:
```js
assert.lengthOf(formatDateLong, 1);
```

Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md:94]: It is not checking specifically for the date object, I have updated the hint text.

File: curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md (hunk 3 above)
Comment (zairahira) [reply curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md:39]: Yes, exactly this is what was required.

File: curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+assert.match(code, /console\.log\(\s*currentDateFormat\s*\)/);
+```
+
+You should have a function `formatDateMMDDYYYY` 
+
+```js
+assert.isFunction(formatDateMMDDYYYY);
+```
+
+The function`formatDateMMDDYYYY` should take a date object as a parameter.
```
Comment (Dario-DC) [inline curriculum/challenges/english/25-front-end-development/lab-date-conversion/66f686b8ebdb982fa8e14330.md:64]:
```suggestion
The function`formatDateMMDDYYYY` should take a single parameter.
```

**Review by Dario-DC — COMMENTED** [review]
Just tested and it works well for me :+1:  Left one comment.

**Review by Dario-DC — APPROVED** [review]

File: curriculum/challenges/_meta/lab-date-conversion/meta.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "Build a Date Conversion Program",
+  "isUpcomingChange": true,
+  "usesMultifileEditor": true,
+  "dashedName": "lab-date-conversion",
+  "order": 219,
+  "superBlock": "front-end-development",
+  "challengeOrder": [{ "id": "66f686b8ebdb982fa8e14330", "title": "Step 1" }],
```
Comment (moT01) [inline curriculum/challenges/_meta/lab-date-conversion/meta.json:8]:
```suggestion
  "challengeOrder": [{ "id": "66f686b8ebdb982fa8e14330", "title": "Build a Date Conversion Program" }],
```

I don't think this gets used or displayed, but might as well fix it.

**Review by moT01 — COMMENTED** [review]
Left one comment. Other than that, LGTM 🎉

**Comment by moT01** [comment]
I went ahead and added my suggestion.

**Review by moT01 — APPROVED** [review]

**Pull request merged**